common/flatpak-transaction.c

```diff
@@ -2061,9 +2169,14 @@ resolve_p2p_ops (FlatpakTransaction *self,
 
       op->download_size = resolve->download_size;
       op->installed_size = resolve->installed_size;
+      op->eol = g_strdup (resolve->eol);
+      op->eol_rebase = g_strdup (resolve->eol_rebase);
 
-      old_metadata_bytes = load_deployed_metadata (self, op->ref);
-      mark_op_resolved (op, resolve->resolved_commit, resolve->resolved_metadata, old_metadata_bytes);
+      if (!check_skip_eol (self, op))
+        {
+          old_metadata_bytes = load_deployed_metadata (self, op->ref);
+          mark_op_resolved (op, resolve->resolved_commit, resolve->resolved_metadata, old_metadata_bytes);
```

Comment:
This is causing issues, later because we're not marking this as resolved, and thus resolve->resolved_commit is NULL causing an assert.

We really need to mark it resolved even if we mark it skip.